File: Multiomics/SNV/Clean/somaticsniper/process.pl
#!/usr/bin/perl -w
use strict;
use warnings;
###################################################################################################
#思路：	对SNV数据进行处理，将33个文件分别读入，并进行计算
		#1.读入33个文件，将其可用信息筛选出来
		#2.将33个文件进行整合，
###################################################################################################

#所有的癌症名
my @cancer_names=(
"BRCA",
"GBM",
"OV",
"LUAD",
"UCEC",
"KIRC",
"HNSC",
"LGG",
"THCA",
"LUSC",
"PRAD",
"SKCM",
"COAD",
"STAD",
"BLCA",
"LIHC",		#i=16
"CESC",
"KIRP",
"SARC",
"LAML",
"PAAD",
"ESCA",
"PCPG",
"READ",
"TGCT",
"THYM",
"KICH",
"ACC",
"MESO",
"UVM",
"DLBC",
"UCS",
"CHOL"
);


for my $k (0 .. 32)
{
	my $snv_path_name="TCGA.".$cancer_names[$k].".somaticsniper.somatic.maf";
	print "$snv_path_name\n";
	
	open SNV, $snv_path_name or die "Can't open SNV";
	my $count=0;
	my @snv;
	while(<SNV>)
	{
		if($count>=5)
		{
			my @temp=split /\s+/, $_ ;
			$snv[$count-5][0]=$temp[0];
			$snv[$count-5][1]=$temp[1];
			$snv[$count-5][2]=$temp[4];
			$snv[$count-5][3]=$temp[5];
			$snv[$count-5][4]=$temp[6];
		}	
		$count=$count+1;
	}
	close(SNV);
	print "count : $count\n";

	my $snv_res_name=">TCGA.".$cancer_names[$k].".somaticsniper.somatic.maf.txt";
	open RESULT,$snv_res_name or die "Can't open RESULT";	
	#print RESULT "ID\tname\n";
	$count=0;
	#计算overlap
	for my $i (0 .. $#snv)
	{
		
		for my $j (0 .. $#{$snv[$i]})
		{
				print RESULT "$snv[$i][$j]\t";
		}
		print RESULT "\n";
	}

}























=cut

File: Multiomics/SNV/Clean/somaticsniper/process.R
cancer_names=c(	#33 cancers
"BRCA",
"GBM",
"OV",
"LUAD",
"UCEC",
"KIRC",
"HNSC",
"LGG",
"THCA",
"LUSC",
"PRAD",
"SKCM",
"COAD",
"STAD",
"BLCA",
"LIHC",		#i=16
"CESC",
"KIRP",
"SARC",
"LAML",
"PAAD",
"ESCA",
"PCPG",
"READ",
"TGCT",
"THYM",
"KICH",
"ACC",
"MESO",
"UVM",
"DLBC",
"UCS",
"CHOL"
)

SNV_ALL=c()

for(i in 1:length(cancer_names))
{
	print(i)
	snv_path_name=paste("TCGA.", cancer_names[i], sep = "", collapse = "")
	snv_path_name=paste(snv_path_name, ".somaticsniper.somatic.maf.txt", sep = "", collapse = "")

	SNV=read.table(file = snv_path_name,
					sep = "\t", header = TRUE, 
					row.names = NULL, stringsAsFactors = FALSE)
	print(nrow(SNV))
	SNV=subset(SNV, select = -X )
	SNV=SNV[order(SNV[,1]),]
	SNV=SNV[!duplicated(SNV$Hugo_Symbol),] #delete the duplicate gene item
	print(nrow(SNV))
	SNV$data_type=i
	SNV_ALL=rbind(SNV_ALL,SNV)	
}
write.table(SNV_ALL, file ="SNV.process.txt",sep="\t",row.names =FALSE, col.names =TRUE, quote =TRUE);


